internal/command/launch/launch.go

```diff
@@ -309,6 +302,22 @@ func run(ctx context.Context) (err error) {
 		fmt.Fprintf(io.Out, "Created app '%s' in organization '%s'\n", appConfig.AppName, org.Slug)
 	}
 
+	if shouldUseMachines {
+		// Check imported fly.toml is a valid V2 config before creating the app
+		if err := appConfig.SetMachinesPlatform(); err != nil {
+			return fmt.Errorf("Can not use configuration for Apps V2, check fly.toml: %w", err)
+		}
+
+		// remove auto-rollback from machine fly.tomls
+		appConfig.Experimental = nil
```

Comment:
There is a subtle difference on nullifying the experimental section here instead of where it was before.

Before it was after the call to `setAppconfigFromSrcinfo()` which among other things sets some experimental features like docker entrypoint and docker command.

Honestly, I still think your change is better because we initial intention for experimental was to remove "AutoRollback" sent by Web when fetching app config for a new app. In summary, I'd move it to https://github.com/superfly/flyctl/blob/42c0ac74d8b687ac16e4e6d111300466d0c886cb/internal/command/launch/launch.go#L295-L297 so new apps won't get experimental by default but existent doesn't remove it silently.

Additionally, we probably need a warning for `fly deploy` to v2 apps when experimental features are set.

Reply:
Ok. I'll make this change now.